snap/hooks/install

```diff
@@ -2,33 +2,27 @@
 
 set -eux
 
-function set_default_properties () {
-    for DIR in log-data logs conf opt/zookeeper;
-    do
-        mkdir -p "${SNAP_COMMON}"/$DIR
-        chmod -R 774 "${SNAP_COMMON}"/$DIR
-        chown -R snap_daemon "${SNAP_COMMON}"/$DIR
-    done
-
-    ls -lah "${SNAP_COMMON}"
-
-    cp "${SNAP}"/log4j.properties "${SNAP_COMMON}"/conf/log4j.properties
-    cp -rL "${SNAP}"/usr/lib/jvm/ "${SNAP_COMMON}"/
-    cp "${SNAP}"/jmx_prometheus_javaagent.jar "${SNAP_COMMON}"/opt/zookeeper/jmx_prometheus_javaagent.jar
-    cp "${SNAP}"/jmx_prometheus.yaml "${SNAP_COMMON}"/conf/jmx_prometheus.yaml
-
-    ls -lah "${SNAP_COMMON}"
-    ls -lah "${SNAP_COMMON}"/jvm
-
-    chmod -R 774 ${SNAP_COMMON}/conf/log4j.properties
-    chmod -R 774 ${SNAP_COMMON}/jvm
-    chmod -R 774 ${SNAP_COMMON}/opt/zookeeper/jmx_prometheus_javaagent.jar
-    chmod -R 774 ${SNAP_COMMON}/conf/jmx_prometheus.yaml
-
-    chown -R snap_daemon ${SNAP_COMMON}/conf/log4j.properties
-    chown -R snap_daemon ${SNAP_COMMON}/jvm
-    chown -R snap_daemon ${SNAP_COMMON}/opt/zookeeper/jmx_prometheus_javaagent.jar
-    chown -R snap_daemon ${SNAP_COMMON}/conf/jmx_prometheus.yaml
-}
-
-set_default_properties
+export COMPONENT=zookeeper
+
+export conf="${SNAP_DATA}"/etc/$COMPONENT
+export logs="${SNAP_COMMON}"/var/lib/$COMPONENT
+export data="${SNAP_COMMON}"/var/log/$COMPONENT
+export binaries="${SNAP}"/opt/$COMPONENT
+
+mkdir -p $conf
+mkdir -p $logs
+mkdir -p $data
+mkdir -p $binaries
+
+cp "${SNAP}"/log4j.properties $conf
```

Comment:
Although configurations should go in SNAP_DATA, I believe it makes sense to have some defaults (like `log4j.properties` and `jmx_prometheus.yaml`) in the SNAP read-only fs. defaults that are copied and pasted at installation, if files do not yet exist. 

So, how about having a folder in the SNAP directory (thus read only) which we copy and paste data on install hook. I mean having a folder like `${SNAP}/etc/defaults/${COMPONENT}` and in the install hook we do 

`for ITEM in $(ls ${SNAP}/etc/defaults/); do if  [ ! -e "${SNAP_DATA}/etc/$ITEM"]; then cp -r $ITEM "${SNAP_DATA}/etc/$ITEM"; fi; done`

This should initialize the $SNAP_DATA and create a representation of defaults there

Reply:
I do like the idea of organizing everything into opt on the snap build process side of things.  This way the files are already where they need to be when rockcraft copies the snap filesystem over to the rock.

Reply:
@deusebio - I believe **all** configuration files, default or not, should be within the `$conf` directory. When an admin needs to dig around to see what config is running, they have a single point of entry.

Reply:
@deusebio - `jmx_prometheus.yaml`, `log4j.properties` and `with_config.sh` are all within the `local` dir in the snap, which will be automatically put into `$SNAP_DATA`.

As the config is going into `$SNAP_DATA/etc/zookeeper`, I've moved the default config in `local` to `local/etc/zookeeper`, and updated the `install` hook to move those default files to the expected `$SNAP_DATA/etc/zookeeper`

I hope this addresses this point.

Reply:
yes, it does. Thanks! Resolving conversation!